src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/single-table-design/index.mdx

```diff
@@ -0,0 +1,486 @@
+  BillingMode,
+  ProjectionType,
+  Table,
+} from "aws-cdk-lib/aws-dynamodb";
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+const racingTable = new Table(Stack.of(backend.data), "RacingTable", {
+  tableName: "RacingTable",
+  partitionKey: {
+    name: "pk",
+    type: AttributeType.STRING,
+  },
+  sortKey: {
+    name: "sk",
+    type: AttributeType.STRING,
+  },
+  billingMode: BillingMode.PAY_PER_REQUEST,
+});
+
+// LSI: Partition key: pk, Sort key: numeric
+racingTable.addLocalSecondaryIndex({
+  indexName: "LocalSecondaryIndex",
+  sortKey: { name: "numeric", type: AttributeType.NUMBER },
+  projectionType: ProjectionType.ALL,
+});
+
+// GSI: Partition key: sk, Sort key: numeric
+racingTable.addGlobalSecondaryIndex({
+  indexName: "GlobalSecondaryIndex",
+  partitionKey: { name: "sk", type: AttributeType.STRING },
+  sortKey: { name: "numeric", type: AttributeType.NUMBER },
+});
+
+backend.data.addDynamoDbDataSource("RacingTableDataSource", racingTable);
+```
+</Block>
+</BlockSwitcher>
+
+## Using custom mutations & queries
+
+To interact with the external DynamoDB table, you can add custom types, mutations, and queries to the `amplify/data/resource.ts` file.
+
+Here's an example schema for the racing application:
+
+```ts
+import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
+
+const schema = a.schema({
+  RacingTable: a.customType({
+    pk: a.string().required(),
+    sk: a.string().required(),
+    numeric: a.float(),
+    results: a.json(),
+  }),
+  createRace: a
+    .mutation()
+    .arguments({
+      raceId: a.string().required(),
+      classId: a.string(),
+      results: a.json(),
+    })
+    .returns(a.ref("RacingTable"))
+    .handler(
+      a.handler.custom({
+        dataSource: "RacingTableDataSource",
+        entry: "./createRace.js",
+      })
+    )
+    .authorization((allow) => [allow.publicApiKey()]),
+  getRaceResultsByRacerId: a
+    .query()
+    .arguments({
+      racerId: a.string().required(),
+    })
+    .returns(a.ref("RacingTable").array())
+    .handler(
+      a.handler.custom({
+        dataSource: "RacingTableDataSource",
+        entry: "./getRaceResultsByRacerId.js",
+      })
+    )
+    .authorization((allow) => [allow.publicApiKey()]),
+  getRacesByClassId: a
+    .query()
+    .arguments({
+      classId: a.string().required(),
+    })
+    .returns(a.ref("RacingTable").array())
+    .handler(
+      a.handler.custom({
+        dataSource: "RacingTableDataSource",
+        entry: "./getRacesByClassId.js",
+      })
+    )
+    .authorization((allow) => [allow.publicApiKey()]),
+  getBestPerformanceByClassId: a
+    .query()
+    .arguments({
+      classId: a.string().required(),
+    })
+    .returns(a.ref("RacingTable").array())
+    .handler(
+      a.handler.custom({
+        dataSource: "RacingTableDataSource",
+        entry: "./getBestPerformanceByClassId.js",
+      })
+    )
+    .authorization((allow) => [allow.publicApiKey()]),
```

Comment:
why not just `.public()`? In general I don't like the idea of using api keys in our examples / docs because they're a time bomb that causes customer projects to break in the future. Since the default `npx create amplify` includes the auth category, I think it's safe to use everywhere in our docs.